### PR TITLE
[Merged by Bors] - Remove unused deps

### PR DESCRIPTION
**Pull request by bjorn3**
# Objective

Reduce compilation time

# Solution

Remove unused dependencies. While this PR doesn't remove any crates from `Cargo.lock`, it may unlock more build parallelism.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_tasks/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,6 @@ futures-lite = "1.4.0"
 event-listener = "2.4.0"
 async-executor = "1.3.0"
 async-channel = "1.4.2"
-instant = { version = "0.1", features = ["wasm-bindgen"] }
```
Comment (mockersf) [inline crates/bevy_tasks/Cargo.toml:23]: this one is used in a test, maybe it can be moved to a build dependency?

**Comment by NathanSWard** [comment]
Is there any Github action we can add to make sure we don't have unused dependencies?

**Comment by bjorn3** [comment]
`cargo udeps` is a useful command for this.

File: crates/bevy_tasks/Cargo.toml (hunk 1 above)
Comment (bjorn3) [reply crates/bevy_tasks/Cargo.toml:23]: Added it as as dev-dependency.

**Comment by bjorn3** [comment]
The duplicate dependency skip list needs to be updated it seems. proc-macro-crate is now duplicated.

**Comment by mockersf** [comment]
yup #2456

**Review by mockersf — APPROVED** [review]

**Comment by bjorn3** [comment]
Rebased to re-trigger CI.

**Pull request closed without merging**